control-plane/api-gateway/gatekeeper/dataplane.go

```diff
@@ -105,18 +105,15 @@ func consulDataplaneContainer(config common.HelmConfig, gcc v1alpha1.GatewayClas
 		container.Resources = *gcc.Spec.DeploymentSpec.Resources
 	}
 
-	// If running in vanilla K8s, run as root to allow binding to privileged ports;
-	// otherwise, allow the user to be assigned by OpenShift.
 	container.SecurityContext = &corev1.SecurityContext{
-		ReadOnlyRootFilesystem: pointer.Bool(true),
-		// Drop any Linux capabilities you'd get as root other than NET_BIND_SERVICE.
+		AllowPrivilegeEscalation: pointer.Bool(false),
```

Comment:
Setting this will force practitioners to use the `mapPrivilegedContainerPorts` setting that we added recently rather than binding to a privileged port such as `80` which works prior to this change.

Reply:
One thing to consider: Should we change the default value of `mapPrivilegedContainerPorts` to something other than `0`?

Reply:
Do we know what other people tend to do by default?